whimsical-woodpeckers/www/src/py/components/directory/__init__.py

```diff
@@ -0,0 +1,86 @@
+from pyvue import Component
+from components.directory.user import User
+from datetime import datetime
+from fetch import Fetch
+from websocket import WebSocket
+
+# __pragma__ ('skip')
+console = window = this = parent_this = JSON = 0
+# __pragma__ ('noskip')
+
+
+users = [
+    {"name": "John Doe", "last_online": datetime.now().strftime("%B %d, %Y , %H:%M"), "id": 1}
+
+]
+
+
+class Directory(Component):
+    data = {
+        'users': users,
+    }
+    components = {
+        "user": User.get_component()
+    }
+
+    template = "#directory-template"
+
+    props = ['mode']
+
+    @staticmethod
+    def created():
+        print("help")
+        this.create_socket()
+
+    @staticmethod
+    def create_socket():
+        # Get info about our authentication
+        this.id = 1234
+        this.token = 1234
+        parent_this = this
+        parent_this.callbacks = 2
+
+        def callback():
+            parent_this.callbacks -= 1
+            if parent_this.callbacks == 0:
+                parent_this.socket = MessageSocket(lambda parent_this: parent_this.handle_message(this),
+                                                   parent_this.token, parent_this.id)
+
+        def data_callback(data):
+            parent_this.id = data['id']
+            callback()
+
+        def auth_callback(data):
+            parent_this.token = data['token']
+            callback()
+
+        Fetch.get("data/") \
+            .then(lambda response: response.json()) \
+            .then(lambda data: data_callback(data))
+        Fetch.get("auth/") \
+            .then(lambda response: response.json()) \
+            .then(lambda data: auth_callback(data))
```

Comment:
This looks very overengineered. Method chaining using lambdas. This is either brilliant or terrible. I'm not really sure what. ill edge on it being brilliant.